addon/metrics-adapters/m-particle.js

```diff
@@ -0,0 +1,148 @@
+import objectTransforms from "../utils/object-transforms";
+import BaseAdapter from "ember-metrics/metrics-adapters/base";
+const { compact } = objectTransforms;
+import { assert } from "@ember/debug";
+import { assign } from "@ember/polyfills";
+import removeFromDOM from "ember-metrics/utils/remove-from-dom";
+
+export default BaseAdapter.extend({
+  toStringExtension() {
+    return "mParticle";
+  },
+
+  init() {
+    const config = assign({}, this.config);
+    const { apiKey, options } = config;
+    assert(
+      `[ember-metrics] You must pass a valid \`apiKey\` to the ${this.toString()} adapter`,
+      apiKey
+    );
+
+    window.mParticle = {
```

Comment:
Is everything from here on down provided by mParticle?

Reply:
that's right, it's all part of our snippet to get mParticle onto the page.  I can add that as a comment